### PR TITLE
Fixes for variable wsrep_on

### PR DESCRIPTION
**Pull request by sciascid**
Two commits:
1) Fix wsrep.variables_debug. This is a followup related wsrep_provider read-only.
2) Fixes for variable wsrep_on

### CODE REVIEW

File: sql/wsrep_trans_observer.h [hunk 1]
```diff
@@ -433,6 +433,16 @@ static inline void wsrep_close(THD* thd)
   DBUG_VOID_RETURN;
 }
 
+static inline void wsrep_cleanup(THD* thd)
+{
```
Comment (sjaakola) [inline sql/wsrep_trans_observer.h:437]: would it be sufficient to do this client state cleanup as part of wsrep_close() ?
If not, I suggest changing the name of this function to narrow down to client state cleanup only, e.g.: wsrep_client_state_cleanup

**Review by sjaakola — COMMENTED** [review]
some minor comments to take a look at

File: sql/wsrep_trans_observer.h (hunk 1 above)
Comment (sciascid) [reply sql/wsrep_trans_observer.h:437]: It's best to leave wsrep_cleanup() as is. There reason is that we do have wsrep_cleanup() in mysql-wsrep 8.0. I think it is better to leave them the same.

File: mysql-test/suite/wsrep/t/variables_debug.test [hunk 2]
```diff
@@ -23,7 +23,7 @@ SET GLOBAL wsrep_replicate_myisam= ON;
 
 # Reset it back.
 SET GLOBAL wsrep_replicate_myisam= OFF;
-SET GLOBAL wsrep_provider=none;
+#SET GLOBAL wsrep_provider=none;
```
[the remark this answers is not in the document]
Comment (sciascid) [reply mysql-test/suite/wsrep/t/variables_debug.test:26]: fixed

**Pull request closed without merging**